data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -155,28 +154,33 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
         Map<Buffer, CompletableFuture<InvokeResponse>> bufferToFutureMap = LambdaCommonHandler.sendRecords(
             recordsToLambda, lambdaProcessorConfig, lambdaAsyncClient,
             new OutputCodecContext());
-
         for (Map.Entry<Buffer, CompletableFuture<InvokeResponse>> entry : bufferToFutureMap.entrySet()) {
             CompletableFuture<InvokeResponse> future = entry.getValue();
             Buffer inputBuffer = entry.getKey();
             try {
                 InvokeResponse response = future.join();
                 Duration latency = inputBuffer.stopLatencyWatch();
                 lambdaLatencyMetric.record(latency.toMillis(), TimeUnit.MILLISECONDS);
+                requestPayloadMetric.record(inputBuffer.getPayloadRequestSize());
                 if (isSuccess(response)) {
+                    resultRecords.addAll(convertLambdaResponseToEvent(inputBuffer, response));
                     numberOfRecordsSuccessCounter.increment(inputBuffer.getEventCount());
                     numberOfRequestsSuccessCounter.increment();
-                    resultRecords.addAll(convertLambdaResponseToEvent(inputBuffer, response));
+                    if (response.payload() != null) {
+                        responsePayloadMetric.record(response.payload().asByteArray().length);
+                    }
+                    continue;
```

Comment:
Please remove this `continue` and use the catch to increment. This approach is likely to lead to future code bugs.